include/irr/asset/filters/CSwizzleAndConvertImageFilter.h

```diff
@@ -150,11 +177,52 @@ class CSwizzleAndConvertImageFilter : public CImageFilter<CSwizzleAndConvertImag
 					for (auto blockX=0u; blockX<blockDims.x; blockX++)
 					{
 						auto localOutPos = readBlockPos*blockDims+commonExecuteData.offsetDifference;
-						uint8_t* dstPix = commonExecuteData.outData+commonExecuteData.oit->getByteOffset(localOutPos,commonExecuteData.outByteStrides);
-						if constexpr(std::is_base_of<PolymorphicSwizzle,Swizzle>::value)
-							convertColor<inFormat,outFormat,Swizzle>(srcPix,dstPix,blockX,blockY,state->swizzle);
-						else
-							convertColor<inFormat,outFormat,Swizzle>(srcPix,dstPix,blockX,blockY,*state);
+						uint8_t* dstPix = commonExecuteData.outData+commonExecuteData.oit->getByteOffset(localOutPos + core::vectorSIMDu32(blockX, blockY),commonExecuteData.outByteStrides);
+						
+						auto getSwizzle = [&]() -> Swizzle&
+						{
+							if constexpr (std::is_base_of<PolymorphicSwizzle, Swizzle>::value)
+								return state->swizzle;
+							else
+								return *state;
+						};
+
+						const Swizzle& swizzle = getSwizzle();
+
+						constexpr auto maxChannels = 4;
+						uint64_t decodeBuffer[maxChannels] = {};
+						uint64_t encodeBuffer[maxChannels] = {};
+
+						asset::decodePixels<inFormat>(srcPix, reinterpret_cast<decodeBufferType*>(decodeBuffer), blockX, blockY);
+						swizzle.template operator()<void, void>(decodeBuffer, encodeBuffer);
```

Comment:
see how swizzle gets implemented?

Reply:
?

Reply:
this is the "correct" way to implement Swizzle

Reply:
okay